flaml/autogen/agent/agent.py

```diff
@@ -30,22 +31,95 @@ def _remember(self, memory):
         """Remember something."""
         self._memory.append(memory)
 
-    def _send(self, message, recipient):
+    @staticmethod
+    def _message_to_dict(message: Union[Dict, str]):
+        """Convert a message to a dictionary.
+
+        The message can be a string or a dictionary. The string with be put in the "content" field of the new dictionary.
+        """
+        if isinstance(message, str):
+            return {"content": message}
+        else:
+            return message
+
+    def _append_oai_message(self, message: Union[Dict, str], role, conversation_id):
+        """Append a message to the openai conversation.
+
+        If the message received is a string, it will be put in the "content" field of the new dictionary.
+        If the message received is a dictionary and does not have any of the two fields "content" or "function_call",
+            the "content" field will be set to empty string for valid openai message.
+
+        Args:
+            message (dict or str): message to be appended to the openai conversation.
+            role (str): role of the message, can be "assistant" or "function".
+            conversation_id (str): id of the conversation, should be the name of the recipient or sender.
+        """
+        message = self._message_to_dict(message)
+        # create openai message to be appended to the openai conversation that can be passed to oai directly.
+        oai_message = {k: message[k] for k in ("content", "function_call", "name") if k in message}
+        if "content" not in oai_message and "function_call" not in oai_message:
```

Comment:
I think when this condition is True, we shouldn't append the message to self._oai_conversations because it's not a oai message.